sw/host/provisioning/ft_lib/src/lib.rs

```diff
@@ -168,6 +169,44 @@ pub enum KeyWrapper {
     CkmsKey(String),
 }
 
+// Types of objects which can come from the device in the perso blob.
+#[repr(C)]
+#[derive(Copy, Clone, PartialEq, Eq)]
+enum ObjType {
+    UnendorsedX509Cert = perso_objects::perso_object_type_kPersoObjectTypeX509Tbs as isize,
+    EndorsedX509Cert = perso_objects::perso_object_type_kPersoObjectTypeX509Cert as isize,
+    DevSeed = perso_objects::perso_object_type_kPersoObjectTypeDevSeed as isize,
+}
+
+impl ObjType {
+    fn from_usize(value: usize) -> Result<ObjType> {
+        match value {
+            0 => Ok(ObjType::UnendorsedX509Cert),
+            1 => Ok(ObjType::EndorsedX509Cert),
+            3 => Ok(ObjType::DevSeed),
+            _ => bail!("incorrect input value of {value} for ObjType"),
+        }
+    }
+}
+
+// LTV header of the transferred object, on the wire packed in u16.
+type ObjHeaderType = perso_objects::object_header_t;
+struct ObjHeader {
+    obj_size: usize,
+    obj_type: ObjType,
+}
+
+// Header of the certificate payload of the LTV object, on the wire packed in
+// u16.
+type CertHeaderType = perso_objects::cert_header_t;
+struct CertHeader<'a> {
+    // Total size the certificate takes in the buffer, header + hame length +
+    // cert size.
+    wrapped_size: usize,
+    cert_name: &'a str,
+    cert_body: Vec<u8>,
+}
+
```

Comment:
Convention in the OpenTitan repo so far is to group all of the bindgen'd assets into a separate lib and then import that lib where it is used. See an example of the otp_ctrl DIF bindgen'd lib: https://cs.opensource.google/opentitan/opentitan/+/master:sw/host/opentitanlib/src/dif/otp_ctrl.rs;l=1 . I think we should do the same here so this file does not get bloated with a bunch of definitions of object types and enums. This lib is supposed be simply a set of functions required to drive an FT flow from the host side.

Reply:
done(I think)